plugins/outputs/dynatrace/dynatrace.go

```diff
@@ -219,6 +232,15 @@ func (d *Dynatrace) Write(metrics []telegraf.Metric) error {
 				default:
 					fmt.Fprintf(&buf, "%s%s %v\n", metricID, tagb.String(), value)
 				}
+
+				if metricCounter%1000 == 0 {
```

Comment:
Shouldn't `1000` be configurable or at least not used as a magic number?

Reply:
This is to ensure that the bulks sent to the api are not too large (that's also the reason, why it isn't configurable)

Reply:
keep in mind if this splits it into chunks and the last chunk fails, all chunks will be retried(re-sent) when returning an error. When returning a nil, no chunks will be retried(re-sent).
I wonder if instead we should allow plugins to request that the batch size not exceed a certain limit.

Reply:
Since that isn't possible now, I think the solution here is sufficient. 1000 is a hard limit set by the dynatrace api rate limiting so a configuration wouldn't do any good.

Retrying the whole batch if the last chunk fails is unfortunate and something i'm taking a note of that I might fix in a follow up. Thanks for the heads-up.

Reply:
Hey @zak-pawel! You still have a _request changes_ in place for this comment. Would you be fine with merging this now? Thanks.

Reply:
@arminru I understand why it should not be configurable. 
But I don't understand why this magic number cannot be converted to const. You can name it `dynatraceAPIRateLimit` (or similar) and everyone will be happy :)

Reply:
Sure, making it a const for better readability makes sense - will do!

Reply:
@arminru Now it's great.

Reply:
Thanks!
@ssoroka can this be merged now? 🙂

Reply:
Thanks!
@ssoroka can this be merged now? 🙂